pandas/tests/test_index.py

```diff
@@ -368,6 +368,11 @@ def test_difference_base(self):
                 with tm.assertRaisesRegexp(TypeError, msg):
                     result = first.difference([1, 2, 3])
 
+            # GH 10596 - empty difference retains index's type
+
+            result = idx.difference(idx)
+            self.assertIsInstance(result, type(idx))
```

Comment:
so the way to tthis generically is define a ``create_empty`` method analagous to ``create_index`` for each subclass which has is an empty version of ``create_index``.